services/provider/server/server.go

```diff
@@ -668,10 +669,32 @@ func (s *OCSProviderServer) GetStorageClaimConfig(ctx context.Context, req *pb.S
 			return nil, status.Error(codes.Internal, msg)
 		}
 	}
+
+	// fetch storage cluster peer to indicate whether replication is enabled
+	scp := &ocsv1.StorageClusterPeerList{}
+	if err = s.client.List(ctx, scp, client.InNamespace(s.namespace)); err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to get StorageClusterPeerList. %v", err)
+	}
+	replicationEnabled := len(scp.Items) > 1
+	ID := ""
+	if replicationEnabled {
+		storageClaimName, err := json.Marshal(req.StorageClaimName)
+		if err != nil {
+			klog.Errorf("failed to marshal a name for a storage claim request based on %v. %v", s, err)
+			panic("failed to marshal storage class request name")
+		}
+		md5Sum := md5.Sum(storageClaimName)
+		ID = hex.EncodeToString(md5Sum[:16])
+	} // todo storageID in provider is planned to be string(r.storageClaim.UID)
+	// SID for RamenDR
+	SID := req.StorageConsumerUUID
```

Comment:
StorageID is used to match V*C to StorageClass, we don't need it to be storageClaim UID, we can have it as storageRequest UID as well.

Reply:
We've already agreed that it'll be `string(r.storageClaim.UID)` in [#168](https://github.com/red-hat-storage/ocs-client-operator/pull/168). Eventually the value that will be assigned to this ID is the one from the client, so it actually doesn't matter what value is chosen on this end.

Reply:
As the concept for storageClaim might go away in the future because of convergence, we should pivot and not use it

Reply:
Even so, the final value assigned would be determined on the client.

Reply:
Why do we need to determine the label value on the client if we can do it from the provider itself?

Reply:
We could, I'm saying with the current state of things this is what would happen. After this PR will get merged, the value assigned from here will be propagated to the client.

Reply:
Yes, I agree that we could but I think as storageID is an arbitrary value used by ramen to cross-link the V*RC/V*SC and SC, the hash of storageRequest UID perfectly fits our use case. I don't think we need to change it further on the client. Let's wait for others to review this.